sdks/java/core/src/test/java/org/apache/beam/sdk/transforms/ParDoTest.java

```diff
@@ -4851,6 +4851,342 @@ public void onTimer2(
         return PDone.in(input.getPipeline());
       }
     }
+
```

Comment:
Should we cover the case where the timer becomes eligible and still fires even though it is being cleared in the same bundle?

e.g.
set timer A for 1, set timer B for 2, advance time to 3, have timer A callback clear B, B still fires since it is part of the same bundle

Reply:
I'm not sure how to guarantee that two timers are in the same bundle. Runners can bundle how they please, so simply advancing time to 3 provides no contract that timers A and B will in the same bundle.

Reply:
The test Luke describes still seems useful. I would guess for most runners it will exercise the desired behavior. Since TestStream (by definition) waits for quiescence before processing the next phase runner's will typically bundle everything together because it will be queued up.

Reply:
Ok, but I'll add this test to my followon PR that actually fixes timers when there are in-bundle modifications. The correct test will fail currently.

Reply:
sg